### PR TITLE
Set Logentries level + output pretty logs in test environment

### PR DESCRIPTION
**Pull request by lledey**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -5,25 +5,33 @@ const raven = require('raven');
...
-    stream: process.stdout
-  }],
+  streams: [],
   serializers: {
-    err: bunyan.stdSerializers.err   // <--- use this
+    err: bunyan.stdSerializers.err
   }
 };
 
+if (process.env.NODE_ENV === 'test') {
```
Comment (jean-k-cp) [inline index.js]: you should never do this kind of check. Define another env variable that is set by your test setup file

File: README.md [hunk 2]
```diff
@@ -45,3 +45,8 @@ logger.debug({ field: 1 }, 'debug message');
 logger.trace({ field: 1 }, 'trace message');
 
 ```
+
+## Notes
+
+In test environment (`process.env.NODE_ENV === 'test'`), the logs will be printed already formatted
```
Comment (jean-k-cp) [inline README.md]: no

File: index.js (hunk 1 above)
Comment (lledey) [inline index.js]: Do you have a name to suggest ?

File: test/index.test.js [hunk 3]
```diff
@@ -3,120 +3,174 @@
...
-    process.stdout.write = oldStdoutWrite;
-  });
+  describe('Log functions', () => {
+    let logger;
+    const logs = [];
+
+    before(() => {
+      oldStdoutWrite = process.stdout.write;
+
+      process.stdout.write = function stubStdout(string) {
```
Comment (jean-k-cp) [inline test/index.test.js]: use stubs

File: test/index.test.js [hunk 4]
```diff
@@ -3,120 +3,174 @@
...
+      delete require.cache[require.resolve('../index.js')];
+      const config = rewire('../index.js').__get__('config');
+
+      expect(config.streams).to.have.lengthOf(1);
+      expect(config.streams[0]).to.not.include.keys('type');
+    });
+
+    it('should use only the default stdout stream NODE_ENV equal to test', () => {
+      process.env.NODE_ENV = 'test';
+      delete require.cache[require.resolve('../index.js')];
```
Comment (jean-k-cp) [inline test/index.test.js]: do not add extensions to required files

File: test/index.test.js [hunk 5]
```diff
@@ -3,120 +3,174 @@
...
-      b: {
+    it('should keep objects safe', () => {
+      logger.info({
+        a: 1,
+        b: {
+          c: 1
+        }
+      }, 'hello');
+      const message = logs.shift();
+      expect(message.a).to.be.eql(1);
```
Comment (jean-k-cp) [inline test/index.test.js]: expect(message).to.deep.equal({ a: 1, b: {...} })

File: test/index.test.js [hunk 6]
```diff
@@ -3,120 +3,174 @@
...
+      expect(config.streams[0]).to.have.property('level', 'info');
+      expect(config.streams[0].stream).to.be.instanceOf(PrettyStream);
+    });
+
+    it('should have sentry stream with LOGENTRIES_TOKEN set', () => {
+      process.env.LOGENTRIES_TOKEN = '00000000-0000-0000-0000-000000000000';
+      delete require.cache[require.resolve('../index.js')];
+      const config = rewire('../index.js').__get__('config');
+
+      expect(config.streams).to.have.lengthOf(2);
```
Comment (jean-k-cp) [inline test/index.test.js]: maybe you should check more that the length

File: index.js (hunk 1 above)
Comment (rossille) [inline index.js]: USE_BUNYAN_PRETTY_STREAM ? (on a side note, I think all of this is a lot of trouble to avoir `| bunyan`, but if it works and is fully tested....)

Comment (lledey) [inline index.js]: This is actually better than `| bunyan`, because the pipe swallows mocha's colors

File: README.md (hunk 2 above)
Comment (rossille) [inline README.md]: This explanation would fit well in the configuration section

Comment (lledey) [inline README.md]: done

File: index.js [hunk 7]
```diff
@@ -5,25 +5,33 @@ const raven = require('raven');
...
+  const PrettyStream = require('bunyan-prettystream'); // eslint-disable-line global-require
```
Comment (rossille) [inline index.js]: Why not just respecting the style instead of disabling the check ?

Comment (lledey) [inline index.js]: Moved it at the top of the file and moved `bunyan-prettystream` as a dependency

File: test/index.test.js (hunk 3 above)
Comment (lledey) [inline test/index.test.js]: The problem here is that if we stub `process.stdout.write`, nothing will be output to the console, including mocha

File: test/index.test.js (hunk 5 above)
Comment (lledey) [inline test/index.test.js]: There are more params in `message` which are not really interesting, such as `hostname`, `pid` or `time`

File: test/index.test.js (hunk 3 above)
Comment (jean-k-cp) [inline test/index.test.js]: ok

**Comment by jean-k-cp** [comment]
LGTM

**Pull request merged**